src/dask_awkward/lib/io/io.py

```diff
@@ -460,15 +457,31 @@ def __call__(self, packed_arg):
         )
 
 
+def return_empty_on_raise(
+    fn: Callable,
+    allowed_exceptions: tuple[type[BaseException], ...],
+    backend: BackendT,
+) -> Callable:
+    @functools.wraps(fn)
+    def wrapped(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except allowed_exceptions:
+            return fn.mock_empty(backend)
+
+    return wrapped
```

Comment:
We discussed adding a report mechanism here, so that the partitions can return a `Report` object (that perhaps the IO func can populate).

This would mean at a high level that we don't build an `dak.Array` from the `io_func` applications; rather, we need an unpacking step so that we build two subgraphs that depend upon the `io_func` result.